hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/buffer/BufferAllocator.java

```diff
@@ -14,13 +14,18 @@
  * limitations under the License.
  */
 
-package com.hazelcast.internal.tpc.iobuffer;
+package com.hazelcast.internal.tpc.buffer;
 
-public interface IOBufferAllocator {
+import java.nio.ByteBuffer;
 
-    IOBuffer allocate();
+public interface BufferAllocator<T extends Buffer> {
+    int DEFAULT_BUFFER_SIZE = 4096;
 
-    IOBuffer allocate(int minSize);
+    T allocate();
 
-    void free(IOBuffer buf);
+    T allocate(int minSize);
+
+    void free(T buf);
+
+    void free(ByteBuffer chunk);
```

Comment:
There are no tests for these two methods, if the BufferAllocator is not growing or concurrent.

Reply:
working on that